activity/worker_test.go

```diff
@@ -0,0 +1,340 @@
+package activity
+
+import (
+	"errors"
+	"log"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/awslabs/aws-sdk-go/aws"
+	"github.com/awslabs/aws-sdk-go/gen/swf"
+	"github.com/sclasen/swfsm/fsm"
+	"github.com/sclasen/swfsm/migrator"
+	. "github.com/sclasen/swfsm/sugar"
+)
+
+type MockSWF struct {
+	Activity *swf.ActivityTask
+}
+
+func (MockSWF) RecordActivityTaskHeartbeat(req *swf.RecordActivityTaskHeartbeatInput) (resp *swf.ActivityTaskStatus, err error) {
+	return nil, nil
+}
+func (MockSWF) RespondActivityTaskCanceled(req *swf.RespondActivityTaskCanceledInput) (err error) {
+	return nil
+}
+func (MockSWF) RespondActivityTaskCompleted(req *swf.RespondActivityTaskCompletedInput) (err error) {
+	return nil
+}
+func (MockSWF) RespondActivityTaskFailed(req *swf.RespondActivityTaskFailedInput) (err error) {
+	return nil
+}
+func (m MockSWF) PollForActivityTask(req *swf.PollForActivityTaskInput) (resp *swf.ActivityTask, err error) {
+	return m.Activity, nil
+}
+
+func ExampleActivityWorker() {
+
+	var swfOps SWFOps
+
+	taskList := "aTaskListSharedBetweenTaskOneAndTwo"
+
+	handleTask1 := func(task *swf.ActivityTask, input interface{}) (interface{}, error) {
+		return input, nil
+	}
+
+	handleTask2 := func(task *swf.ActivityTask, input interface{}) (interface{}, error) {
+		return input, nil
+	}
+
+	handler1 := &ActivityHandler{Activity: "one", HandlerFunc: handleTask1}
+
+	handler2 := &ActivityHandler{Activity: "two", HandlerFunc: handleTask2}
+
+	worker := &ActivityWorker{
+		Domain:     "swf-domain",
+		Serializer: fsm.JSONStateSerializer{},
+		TaskList:   taskList,
+		SWF:        swfOps,
+		Identity:   "test-activity-worker",
+	}
+
+	worker.AddHandler(handler1)
+
+	worker.AddHandler(handler2)
+
+	go worker.Start()
+
+}
+
+func ExampleTypedActivityWorker() {
+
+	var swfOps SWFOps
+
+	taskList := "aTaskListSharedBetweenTaskOneAndTwo"
+
+	worker := &ActivityWorker{
+		Domain:     "swf-domain",
+		Serializer: fsm.JSONStateSerializer{},
+		TaskList:   taskList,
+		SWF:        swfOps,
+		Identity:   "test-activity-worker",
+	}
+
+	var activities Activities
+
+	activities = MockActivities{}
+
+	worker.AddHandler(NewActivityHandler("one", activities.Task1))
+
+	worker.AddHandler(NewActivityHandler("two", activities.Task2))
+
+	go worker.Start()
+
+}
+
+type Input1 struct {
+	Data string
+}
+
+type Output1 struct {
+	Data string
+}
+
+type Input2 struct {
+	Data2 string
+}
+
+type Output2 struct {
+	Data2 string
+}
+
+//We define the operations our activity worker will handle in an interface
+//Then it is easy to provide a mocked impl
+type Activities interface {
```

Comment:
This is an example Interface of an activity worker that handles one or more activity types.